src/apps/components/AppDetailsPage/PermissionsCard.tsx

```diff
@@ -12,22 +13,42 @@ type PermissionsCardProps = {
     code: PermissionEnum;
   }> | null;
   loading: boolean;
+  appId: string; // todo wrap with App Context
 } & BoxProps;
 
 export const PermissionsCard: React.FC<PermissionsCardProps> = ({
   permissions,
   loading,
+  appId,
   ...boxProps
 }) => {
+  const [editPermissionDialogOpen, setEditPermissionDialogOpen] =
+    useState(false);
   const intl = useIntl();
 
+  const editPermissionsButton = (
```

Comment:
I understand why it was declared here, but I still think this should be outside the scope of `PermissionsCard`, accepting the `onClick` handler.

If a component is declared inside another component, it will be recreated on each render.

Reply:
Just like the rest of the JSX in this component :) I can extract it but IMO its a nitpick, you can write the same thing for almost every line of HTML. When components are not complex (this is a single <button>), there is not much benefit of extracting components

Reply:
I decided not to move it, this button needs also i18n, so I will have to create another wrapper without a real reason. This component is assigned to `const` just to remove duplication. I don't see any benefit of extracting it